New Resource: `azurerm_data_protection_backup_policy_postgresql_flexible_server` (#26024)
hashicorp/terraform-provider-azurerm: website/docs/r/data_protection_backup_policy_postgresql_flexible_server.html.markdown

}

resource "azurerm_data_protection_backup_policy_postgresql_flexible_server" "example" {
  name                            = "example-backup-policy"
  vault_id                        = azurerm_data_protection_backup_vault.example.id
  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
  time_zone                       = "India Standard Time"

  default_retention_rule {
    life_cycle {
      duration        = "P4M"
      data_store_type = "VaultStore"
    }
  }

  retention_rule {
    name     = "weekly"
    duration = "P6M"
    priority = 20

    criteria {
      absolute_criteria = "FirstOfWeek"
    }
  }

  retention_rule {
    name     = "thursday"
    duration = "P1W"
    priority = 25

    criteria {
      days_of_week           = ["Thursday"]
      scheduled_backup_times = ["2021-05-23T02:30:00Z"]
    }
  }

  retention_rule {
    name     = "monthly"
    duration = "P1D"
    priority = 15

    criteria {
      weeks_of_month         = ["First", "Last"]
      days_of_week           = ["Tuesday"]
      scheduled_backup_times = ["2021-05-23T02:30:00Z"]
    }
  }
}
```

## Arguments Reference

The following arguments are supported:

> * `name` - (Required) The name which should be used for this Backup Policy PostgreSQL Flexible Server. Changing this forces a new resource to be created.

Reviewer: I think this resource is a policy, not a server? If so, please also update other places.
```suggestion
* `name` - (Required) Specifies the name of the Backup Policy for the PostgreSQL Flexible Server. Changing this forces a new resource to be created.
```

Author: Updated

* `vault_id` - (Required) The ID of the Backup Vault where the Backup Policy PostgreSQL Flexible Server should exist. Changing this forces a new resource to be created.

> * `backup_repeating_time_intervals` - (Required) Specifies a list of repeating time interval. It supports weekly back. It should follow `ISO 8601` repeating time interval. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `backup_repeating_time_intervals` - (Required) Specifies a list of repeating time interval. It supports weekly back. It should follow `ISO 8601` repeating time interval format. Changing this forces a new resource to be created.
```

Author: Updated

* `default_retention_rule` - (Required) A `default_retention_rule` block as defined below. Changing this forces a new resource to be created.

* `retention_rule` - (Optional) One or more `retention_rule` blocks as defined below. Changing this forces a new resource to be created.

* `time_zone` - (Optional) Specifies the Time Zone which should be used by the backup schedule. Changing this forces a new resource to be created.

---

A `default_retention_rule` block supports the following:

* `life_cycle` - (Required) A `life_cycle` block as defined below. Changing this forces a new resource to be created.

---

A `retention_rule` block supports the following:

> * `name` - (Required) The name which should be used for this retention rule. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `name` - (Required) Specifies the name of the retention rule. Changing this forces a new resource to be created.
```

Author: Updated
